browser/ui/views/speedreader/reader_mode_toolbar_view.h

```diff
@@ -12,30 +12,46 @@
 #include "ui/views/controls/webview/webview.h"
 
 namespace content {
-class BrowserContext;
+class WebContents;
 }
 
+class Browser;
 class ReaderModeToolbarView : public views::View {
   METADATA_HEADER(ReaderModeToolbarView, views::View)
  public:
-  explicit ReaderModeToolbarView(content::BrowserContext* browser_context);
+  struct Delegate {
+    virtual ~Delegate() = default;
+    virtual void OnReaderModeToolbarActivate(ReaderModeToolbarView* toolbar) {}
+  };
+
+  explicit ReaderModeToolbarView(Browser* browser);
   ~ReaderModeToolbarView() override;
 
   ReaderModeToolbarView(const ReaderModeToolbarView&) = delete;
   ReaderModeToolbarView(ReaderModeToolbarView&&) = delete;
   ReaderModeToolbarView& operator=(const ReaderModeToolbarView&) = delete;
   ReaderModeToolbarView& operator=(ReaderModeToolbarView&&) = delete;
 
+  void SetVisible(bool visible) override;
+
   content::WebContents* GetWebContentsForTesting();
 
   views::View* toolbar() const { return toolbar_.get(); }
 
+  void SetDelegate(Delegate* delegate);
+  void SwapToolbarContents(ReaderModeToolbarView* toolbar);
+
+  void ActivateContents();
+
  private:
   gfx::Size CalculatePreferredSize(
       const views::SizeBounds& available_size) const override;
   void OnBoundsChanged(const gfx::Rect& previous_bounds) override;
+  bool OnMousePressed(const ui::MouseEvent& event) override;
 
   std::unique_ptr<views::WebView> toolbar_;
+  std::unique_ptr<content::WebContents> toolbar_contents_;
```

Comment:
I think we need to re-think this ownership because other toolbar can refer this contents by swap.

Reply:
Both toolbars lives as long as BrowserView alive. I don't know is there a better place to store these contents.